Allow incorrect cert in development.

twifkak opened the pull request:
When `-development` is passed, an invalid SXG cert will only cause a
WARNING.

@shigeki

shigeki commented:
Thanks, I missed it.

If `-development` is intended to work with `--ignore-certificate-errors-spki-list` on Chrome, only cert's validity period checks need to be warned. Other checks of cert's key pair and signing domain in `util.CheckCertificate` should be failed.

I think it is better to  move validity period checks out of `util.CheckCertificate` and allow it if `-developement` such as `util.CanSignHttpExchanges`.

twifkak commented:
@shigeki Thanks for the review.

It's not documented in README.md, but I also use `-development` with `curl ... | dump-signedexchange -verify`.

I moved the validity check from `CheckCertificate` to `CanSignHttpExchanges` (and changed its doc and interface accordingly). I also added support for the `-invalidcert` flag which I forgot. Does this seem reasonable?
